crates/sui-core/src/transaction_manager.rs

```diff
@@ -77,16 +87,26 @@ impl TransactionManager {
             if self.authority_store.effects_exists(&digest)? {
                 continue;
             }
+
+            let valid_effects = self.authority_store.get_valid_effects(&digest)?;
```

Comment:
I think there are two alternatives:
1. Pass effects as function args, same as certificates, to remove any doubt that there can be inconsistency (e.g. effect does not exist), and save the trip to rocksdb.
2. Pass TransactionDigest as function arg. Load cert and effects from db. This seems not as preferable but at least more consistent.